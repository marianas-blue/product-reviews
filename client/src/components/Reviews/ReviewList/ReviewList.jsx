/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-unused-state */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/button-has-type */
/* eslint-disable react/no-unescaped-entities */
/* eslint-disable no-useless-constructor */
/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import { Button, Select, Reviews, Title, Starbox, Space } from './ReviewList.style';

const axios = require('axios');

class ReviewList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reviewers: [],
      reviewTitle: [],
      review: [],
      helpful: [],
      datePosted: [],
      done: false,
    };
  }

  componentDidMount() {
    const fullUrl = document.URL;
    const urlArray = fullUrl.split('/');
    const lastSegment = urlArray[urlArray.length - 1];
    axios.get(`http://ec2-34-211-238-169.us-west-2.compute.amazonaws.com/api/product/${lastSegment}`)
      .then((res) => {
        for (let i = 0; i < res.data.length; i += 1) {
          this.state.reviewers.push(res.data[i].reviewer);
          this.state.reviewTitle.push(res.data[i].review_title);
          this.state.review.push(res.data[i].review);
          this.state.helpful.push(res.data[i].helpful_counter);
          this.state.datePosted.push(res.data[i].created_at);
        }
        this.setState({ done: true });
      });
  }

  render() {
    const { reviewers, reviewTitle, review , helpful, datePosted } = this.state;
    return (
      <div>
        <Title>Showing 1-3 of 453 reviews</Title>
        <div>
          <Select name="reviews">
            <option value="Top Reviews">Top Reviews</option>
            <option value="Most recent">Most recent</option>
          </Select>
        </div>

        <Reviews>
          <div>
            <Starbox>
              <img
                src="https://s3.amazonaws.com/product-reviews-hr110/Icons/avatar.png"
                height="30px"
                width="30px"
                alt="avatar"
                max-width="100%"
                display="block"
              />
              {reviewers[0]}
            </Starbox>
            <Starbox>
              <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
              <Title> {reviewTitle[0]} </Title>
            </Starbox>
            <div> {datePosted[0]} </div>
            <p width="auto">
              { review[0] }
            </p>
            <div>
              {helpful[0]} people found this helpful
            </div>
            <Space>
              <Button>Helpful</Button> |
              <a href="#top" style={{ textDecoration: 'none', color: '#7a7a7a' }}>Comment</a> |
              <a href="#top" style={{ textDecoration: 'none', color: '#7a7a7a' }}>Report abuse</a>
            </Space>
          </div>
        </Reviews>

        <Reviews>
          <Starbox>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/avatar.png" height="30px" width="30px" alt="avatar" />
            {reviewers[1]}
          </Starbox>
          <Starbox>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
            <Title> {reviewTitle[1]} </Title>
          </Starbox>
          <div> {datePosted[1]} </div>
          <p width="auto">
            { review[1] }
          </p>
          <div>
            {helpful[1]} people found this helpful
          </div>
          <Space>
            <Button>Helpful</Button> |
            <a href="#top" style={{ textDecoration: 'none', color: '#7a7a7a' }}>Comment</a> |
            <a href="#top" style={{ textDecoration: 'none', color: '#7a7a7a' }}>Report abuse</a>
          </Space>
        </Reviews>

        <Reviews>
          <Starbox>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/avatar.png" height="30px" width="30px" alt="avatar" />
            {reviewers[2]}
          </Starbox>
          <Starbox>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
            <Title> {reviewTitle[2]} </Title>
          </Starbox>
          <div> {datePosted[2]} </div>
          <p width="auto">
            { review[2] }
          </p>
          <div>
            {helpful[2]} people found this helpful
          </div>
          <Space>
            <Button>Helpful</Button> |
            <a href="#top" style={{ textDecoration: 'none', color: '#7a7a7a' }}>Comment</a> |
            <a href="#top" style={{ textDecoration: 'none', color: '#7a7a7a' }}>Report abuse</a>          
          </Space>
        </Reviews>
      </div>
    );
  }
}

export default ReviewList;
