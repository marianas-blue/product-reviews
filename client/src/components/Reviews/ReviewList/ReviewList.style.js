import styled from 'styled-components';

export const Button = styled.button`
  background: #eff1f3;
  border-radius: 3px;
  border-color: #ADB1B8 #A2A6AC #8D9096;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: inline-block;
  padding: 0;
  text-align: center;
  text-decoration: none!important;
  vertical-align: middle;
  box-sizing: border-box;
  width: 100px;
  height: 29px;
  :hover {
    background: #e7e9ec;
  }
  font-size: 13px;
`;

export const Select = styled.select`
  background: #eff1f3;
  width: 100px;
`;

export const Reviews = styled.div`
  padding: 5px;
  line-height: 20px;
  font-size: 13px;
  max-width: 600px;  
`;

export const Title = styled.button`
font-weight: 700;
font-size: 12px;
line-height: 1.255;
border: none; 
padding: 0;
background-color: transparent;
outline: none;
text-align: left;
`;

export const Starbox = styled.div`
display: flex;
align-items: center;
`;

export const Space = styled.div`
margin: 10px 10px 10px 0px;
`;
