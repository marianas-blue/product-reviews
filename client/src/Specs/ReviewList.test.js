/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import ReviewList from '../components/Reviews/ReviewList/ReviewList';

describe('ReviewList', () => {
  it('should render correctly', () => {
    const component = shallow(<ReviewList />);

    expect(component).toMatchSnapshot();
  });
});
