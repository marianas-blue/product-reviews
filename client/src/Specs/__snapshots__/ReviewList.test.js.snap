// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList should render correctly 1`] = `
<div>
  <styled.button>
    Showing 1-8 of 453 reviews
  </styled.button>
  <div>
    <styled.select
      name="reviews"
    >
      <option
        value="Top Reviews"
      >
        Top Reviews
      </option>
      <option
        value="Most recent"
      >
        Most recent
      </option>
    </styled.select>
  </div>
  <styled.div>
    <div>
      <styled.div>
        <img
          alt="avatar"
          display="block"
          height="30px"
          max-width="100%"
          src="https://s3.amazonaws.com/product-reviews-hr110/Icons/avatar.png"
          width="30px"
        />
        Colleen Marquardt
      </styled.div>
      <styled.div>
        <img
          alt="stars"
          height="25px"
          src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
        />
        <styled.button>
           Quod soluta quas ipsam autem dolor libero dolor qui repudiandae.
        </styled.button>
      </styled.div>
      <div>
        September 18, 2018
      </div>
      <p>
        "Pariatur dolorum voluptate commodi iusto omnis alias sed enim rem. Quis omnis ipsam aliquid dolores. Tempore consequuntur dolor est excepturi. Cum et doloremque aut as Id voluptatem sint. Labore vero perferendis omnis voluptatem., "revtitleiew_": "Quod soluta quas ipsam autem dolor libero dolor qui repudiandae."
      </p>
      <div>
        301 people found this helpful
      </div>
      <styled.div>
        <styled.button>
          Helpful
        </styled.button>
         |
        <a
          href="#top"
          style={
            Object {
              "color": "#7a7a7a",
              "textDecoration": "none",
            }
          }
        >
          Comment
        </a>
         |
        <a
          href="#top"
          style={
            Object {
              "color": "#7a7a7a",
              "textDecoration": "none",
            }
          }
        >
          Report abuse
        </a>
      </styled.div>
    </div>
  </styled.div>
  <styled.div>
    <styled.div>
      <img
        alt="avatar"
        height="30px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/avatar.png"
        width="30px"
      />
      Ms. Calista Buckridge
    </styled.div>
    <styled.div>
      <img
        alt="stars"
        height="25px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
      />
      <styled.button>
         Quod soluta quas ipsam autem dolor libero dolor qui repudiandae.
      </styled.button>
    </styled.div>
    <div>
      August 9, 2018
    </div>
    <p>
      "Pariatur dolorum voluptate commodi iusto omnis alias sed enim rem. Quis omnis ipsam aliquid dolores. Tempore consequuntur dolor est excepturi. Cum et doloremque aut as Id voluptatem sint. Labore vero perferendis omnis voluptatem., "review_title": "Quod soluta quas ipsam autem dolor libero dolor qui repudiandae."
    </p>
    <div>
      107 people found this helpful
    </div>
    <styled.div>
      <styled.button>
        Helpful
      </styled.button>
       |
      <a
        href="#top"
        style={
          Object {
            "color": "#7a7a7a",
            "textDecoration": "none",
          }
        }
      >
        Comment
      </a>
       |
      <a
        href="#top"
        style={
          Object {
            "color": "#7a7a7a",
            "textDecoration": "none",
          }
        }
      >
        Report abuse
      </a>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <img
        alt="avatar"
        height="30px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/avatar.png"
        width="30px"
      />
      Reina Beer
    </styled.div>
    <styled.div>
      <img
        alt="stars"
        height="25px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
      />
      <styled.button>
         Quod soluta quas ipsam autem dolor libero dolor qui repudiandae.
      </styled.button>
    </styled.div>
    <div>
      September 18, 2018
    </div>
    <p>
      "Pariatur dolorum voluptate commodi iusto omnis alias sed enim rem. Quis omnis ipsam aliquid dolores. Tempore consequuntur dolor est excepturi. Cum et doloremque aut as Id voluptatem sint. Labore vero perferendis omnis voluptatem., "review_title": "Quod soluta quas ipsam autem dolor libero dolor qui repudiandae."
    </p>
    <div>
      888 people found this helpful
    </div>
    <styled.div>
      <styled.button>
        Helpful
      </styled.button>
       |
      <a
        href="#top"
        style={
          Object {
            "color": "#7a7a7a",
            "textDecoration": "none",
          }
        }
      >
        Comment
      </a>
       |
      <a
        href="#top"
        style={
          Object {
            "color": "#7a7a7a",
            "textDecoration": "none",
          }
        }
      >
        Report abuse
      </a>
    </styled.div>
  </styled.div>
</div>
`;
